File: Model_Day_4/ReadCSV.py
import csv
from Customer import Customer
class ReadCSV:
    def bacaCSV (self, filename):
        customers = []
        try:
            with open(filename, "r") as file:
                reader = csv.DictReader(file)

                for row in reader:
                    name = row["nama"]
                    age = int(row["umur"])
                    salary = float(row["gaji"])
                    existing_loan = float(row["existing_loan"])

                    cust = Customer(name, age, salary, existing_loan)
                    customers.append(cust)


        except FileNotFoundError:
            print("File not found:", filename)

        return customers

File: Model_Day_4/ReadCSV.java
package Model_Day_4;
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.ArrayList;

public class ReadCSV {
    public static ArrayList<Customer2> processCSVFiles(String filename){
    ArrayList<Customer2> customers = new ArrayList<>();    
    try{
        BufferedReader reader = new BufferedReader(
            new FileReader(filename)
        );
        String line;

        reader.readLine();

        while ((line = reader.readLine()) != null){
            String [] data = line.split(",");
            String name = data [0];
            int age = Integer.parseInt(data[1]);
            double salary = Double.parseDouble(data[2]);

            Customer2 cust = new Customer2(name, age, salary, age);
            customers.add(cust);

        }
        reader.close();
    } catch (IOException e){
        System.out.println("error");
    }
    return customers;
}
}
